src/site/topic22.rst

```diff
@@ -263,6 +263,106 @@ Binary Tree Node
 Linked Binary Tree
 ^^^^^^^^^^^^^^^^^^
 
+* Although there will be no implementation of a general ``BinaryTree``, we can discuss what some specific tree based algorithms
+
+**Size**
+
+* If we have some arbitrary binary tree and do not know it's size (and it has no size field), how would we count the number of elements?
+
+.. code-block:: java
+    :linenos:
+
+    public int size() {
+        return size(root);
+    }
+
+    private int size(Node<T> current) {
+        if (current == null) {
+            return 0;
+        } else {
+            return 1 + size((current.getLeft())) + size(current.getRight());
+        }
+    }
+
+* The idea here is, if the current node exists, the the size of the (sub)tree will be 1 + the size of the left subtree + the size of the right subtree
```

Comment:
This probably makes more sense before the implementation, ie present the idea for the solution, then the code.

Similar for other operations.